include/alp/structures.hpp

```diff
@@ -437,7 +455,18 @@ namespace alp {
 		struct isInstantiable< General, Symmetric > {
 			template< typename ImfR, typename ImfC >
 			static bool check( const ImfR &imf_r, const ImfC &imf_c ) {
-				return ( ( imf_r.n == imf_c.n ) && ( imf_r.s == imf_c.s ) && ( imf_r.b == imf_c.b ) );
+				return ( imf_r.n == imf_c.n );
+			};
+		};
+
+		template<>
+		struct isInstantiable< Symmetric, General > {
+			template< typename ImfR, typename ImfC >
+			static bool check( const ImfR &imf_r, const ImfC &imf_c ) {
+				return (
+					( imf_r.map( imf_r.n - 1 ) <= imf_c.map( 0 ) ) ||
+					( imf_c.map( imf_r.n - 1 ) >= imf_r.map( 0 ) )
```

Comment:
I am not sure that the domain of `imf_r` is necessarily a subset of `imf_c`.  Is this second condition meant for a general block in the lower part? In which case `imf_c.map( imf_c.n - 1 ) <= imf_r.map( 0 )`